### DISCUSSION TITLE
Connection Socket failed in release (RN 69.12)

### DISCUSSION THREAD

**Discussion by Tomas-caula** [socketio/socket.io#4911] [Q&A]
Hi, I'm trying to do a Video Call app with ` "socket.io-client": "^4.1.2",` and it's works perfectly but only in the Debug, when I make the apk my app remains blank. 

` useEffect(() => {
    if (!socketUri || !cookie) {
      return;
    }
    console.log('Connecting to socket', socketUri);
    const s = io(socketUri, {
      forceNew: true,
      withCredentials: true,
      transports: ['websocket', 'polling'],
      extraHeaders: {
        Cookie: `token=${cookie};`,
      },
    });

    const p = new Peer();

    setSocket(s);
    setPeer(p);
    // establishFeed();

    return () => {
      s.disconnect();
      p.destroy();
    };
  }, [socketUri, cookie]);`

Note:  All the all permissions are required previously & android:usesCleartextTraffic="true", too

**Comment by darrachequesne** [comment]
Do you get any error? Any log that could help us?

**Reply by Tomas-caula** [reply]
Hi, sorry for the lateness I could fix it. this happens when I run my app in the debug the connection works perfectly with this code 

`useEffect(() => { 
if (!socketUri || !cookie) { return; } 
console.log('Connecting to socket', socketUri); 
const s = io(socketUri, 
{ 
  forceNew: true, 
  withCredentials: true, 
  transports: ['websocket', 'polling'], 
  extraHeaders: { Cookie:token=${cookie};`,
},
});`

well when I try the same in the release version the data is send it different (twice) so my backend don't catch it, I just remove the extraHeaders and it's works perfectly

**Reply by darrachequesne** [reply]
That's really weird. It seems that the `extraHeaders` option should be supported by the WebSocket implementation provided by React Native.

Reference: https://reactnative.dev/docs/network#websocket-support 

Possibly related: https://github.com/socketio/socket.io-client/issues/1593
